src/main/java/org/opensearch/geospatial/index/common/shape/ShapeConverter.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.index.common.shape;
+
+import static org.apache.commons.lang3.ArrayUtils.toPrimitive;
+
+import java.util.Locale;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.DoubleStream;
+import java.util.stream.IntStream;
+
+import org.apache.lucene.geo.XYLine;
+import org.apache.lucene.geo.XYPolygon;
+import org.opensearch.geometry.Line;
+import org.opensearch.geometry.Polygon;
+import org.opensearch.geometry.Rectangle;
+import org.opensearch.geometry.ShapeType;
+
+// Utility class to convert compatible shapes from opensearch to Lucene
+public class ShapeConverter {
+
+    public static XYLine toXYLine(Line line) {
+        Objects.requireNonNull(line, String.format(Locale.getDefault(), "%s cannot be null", ShapeType.LINESTRING));
+        float[] x = toFloatArray(line.getX());
+        float[] y = toFloatArray(line.getY());
+        return new XYLine(x, y);
+    }
+
+    public static XYPolygon toXYPolygon(Rectangle rectangle) {
+        Objects.requireNonNull(rectangle, "Rectangle cannot be null");
+        // build polygon by assigning points in Counter Clock Wise direction (default for polygon) and end at where
+        // you started since it has to be linear ring
+        // (minX,minY) -> (maxX, minY) -> (maxX, maxY) -> (minX, maxY) -> (minX, minY)
```

Comment:
Is this counter clockwise or clockwise? This example seem to be clock wise to me

Reply:
It is counter clockwise
```
(minX, maxY)<-------(maxX, maxY)
      |                  ↑
      |                  | 
      |                  | 
      ↓                  |
(minX, minY)------->(maxX, minY)

Reply:
oh my bad. I mixed up X,Y axis :( Thanks for explanation @naveentatikonda